server/cluster/cluster.go

```diff
@@ -1690,30 +1693,59 @@ func (c *RaftCluster) SetMinResolvedTS(storeID, minResolvedTS uint64) error {
 	return c.putStoreLocked(newStore)
 }
 
-func (c *RaftCluster) runMinResolvedTSJob(saveInterval time.Duration) {
+func (c *RaftCluster) isLarger(minResolvedTSRealtime uint64) bool {
+	c.RLock()
+	defer c.RUnlock()
+	return minResolvedTSRealtime != math.MaxUint64 && minResolvedTSRealtime > c.minResolvedTS
+}
+
+func (c *RaftCluster) runMinResolvedTSJob() {
+	defer logutil.LogPanic()
 	defer c.wg.Done()
-	if saveInterval == 0 {
-		return
+
+	interval := c.opt.GetMinResolvedTSPersistenceInterval()
+	if interval == 0 {
+		interval = DefaultMinResolvedTSPersistenceInterval
 	}
-	defer logutil.LogPanic()
-	ticker := time.NewTicker(saveInterval)
+	ticker := time.NewTicker(interval)
 	defer ticker.Stop()
+
+	c.LoadMinResolvedTS()
 	for {
 		select {
 		case <-c.ctx.Done():
 			log.Info("min resolved ts background jobs has been stopped")
 			return
 		case <-ticker.C:
-			minResolvedTS := c.GetMinResolvedTS()
-			if minResolvedTS != math.MaxUint64 {
-				c.Lock()
-				c.storage.SaveMinResolvedTS(minResolvedTS)
-				c.Unlock()
+			interval = c.opt.GetMinResolvedTSPersistenceInterval()
+			if interval != 0 {
+				minResolvedTSRealtime := c.GetMinResolvedTS()
+				if c.isLarger(minResolvedTSRealtime) {
```

Comment:
It seems the function `GetMinResolvedTS` and `isLarger` are used together and only here, could we merge these two functions into one, so we just need to c.RLock&RUnlock 1 time? 
Also， I would like to move the L1722 - L1727 into one function for the same reason, there are too many `lock` & `unlock`, maybe we could make the logic easier(reduce a bit performance but friendly to our maintainers), how about calling the following function instead(as an example)?

```
func (c *RaftCluster) checkAndUpdateMinResovedTS() {
	c.Lock()
	defer c.Unlock()

	if !c.isInitialized() {
		return
	}
	curMinResolvedTS := uint64(math.MaxUint64)
	for _, s := range c.GetStores() {
		if !core.IsAvailableForMinResolvedTS(s) {
			continue
		}
		if curMinResolvedTS > s.GetMinResolvedTS() {
			curMinResolvedTS = s.GetMinResolvedTS()
		}
	}
	if curMinResolvedTS == math.MaxUint64 || curMinResolvedTS <= c.minResolvedTS {
		return
	}
	c.minResolvedTS = curMinResolvedTS
	c.storage.SaveMinResolvedTS(curMinResolvedTS)
}
```